conans/test/integration/package_id/compatible_test.py

```diff
@@ -279,7 +279,7 @@ def package_info(self):
 
         client.save({"conanfile.py": GenConanfile().with_require("pkg/0.1")})
         client.run("lock create . -s os=Windows --lockfile-out=deps.lock")
-        client.run("install . --lockfile=deps.lock")
+        client.run("install . -s os=Windows --lockfile=deps.lock")
```

Comment:
Unrelated to the PR, but this was failing - Change needed to pass on my local Macos machine

Reply:
This is slightly concerning why CI is not detecting this on the Mac build, the ``release/2.0`` branch builds runs ``integration`` in Mac too

Reply:
There is something off here. In my branch, there is already `` client.run("install . -s os=Windows --lockfile=deps.lock")`` in this line 😕

Reply:
@franramirez688 also confirms this is passing for in in ``release/2.0``, maybe need to rebase.